test-integ/peering_commontopo/sharedtopology_test.go

```diff
@@ -0,0 +1,82 @@
+package peering
+
+import (
+	"flag"
+	"testing"
+)
+
+// Tests that use commonTopo should implement sharedTopoSuite.
+//
+// Tests that use commonTopo are either cooperative or non-cooperative. Non-cooperative
+// uses of commonTopo include is anything that may interfere with other tests, namely
+// mutations, such as:
+// - any calls to commonTopo.Relaunch; this is generally disruptive to other tests
+// - stopping or disabling nodes
+// - ...
+//
+// Cooperative tests should just call testFuncMayReuseCommonTopo() to ensure they
+// are run in the correct `sharetopo` mode. They should also ensure they are included
+// in the commonTopoSuites slice in TestSuitesOnSharedTopo.
+type sharedTopoSuite interface {
+	testName() string
+	setup(*testing.T, *commonTopo)
+	test(*testing.T, *commonTopo)
+}
+
+var flagNoShareTopo = flag.Bool("no-share-topo", false, "do not share topology; run each test in its own")
```

Comment:
I think the sentence here is incomplete/incorrect `do not share topology; run each test in its own`. Also `-no-share-topo` sounds wonky. Perhaps `disable-topo-sharing`,  `separate-topo`, `isolated-topo` as alternatives

Reply:
> do not share topology; run each test in its own

It should be read "do not share topology; run each test in its own [topology]". I can make that explicit

```suggestion
var flagNoShareTopo = flag.Bool("no-share-topo", false, "do not share topology; run each test in its own isolated topology")
```

> Also -no-share-topo sounds wonky

Yeah, but for CLI args it's a pretty common convention, e.g. [`consul acl auth-method update -no-merge`](https://developer.hashicorp.com/consul/commands/acl/auth-method/update#no-merge) and a bunch of `git` commands like `--no-edit`